### PR TITLE
Terminal: add shortcuts for regularly used commands

### PR DESCRIPTION
**Pull request by smartscripts-nl**
Terminal: add code to store sets of commands as shortcuts, which can then later be executed with one click. These shortcuts can also be edited.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/6607)

### CODE REVIEW

**Comment by smartscripts-nl** [comment]
I see my PR results in conflicts. Is that because it is branched from master? Should I branch from upstream/master?

**Comment by Frenzie** [comment]
Whatever you do, it's always recommended to make sure you're starting from the most up-to-date code. Branching from upstream/master is a good way to do that. ^_^

**Comment by poire-z** [comment]
Probably. Or just `git rebase master` to get up to date with your #6599 that has already been merged.
And before bringing more PRs (that all start from master before #6599), can you finish your 2 other ones #6598 and #6600 please.

File: plugins/terminal.koplugin/main.lua [hunk 1]
```diff
@@ -1,56 +1,381 @@
...
 local Font = require("ui/font")
 local InfoMessage = require("ui/widget/infomessage")
 local InputDialog = require("ui/widget/inputdialog")
+local LuaSettings = require("luasettings")
+local Menu = require("ui/widget/menu")
 local TextViewer = require("ui/widget/textviewer")
 local Trapper = require("ui/trapper")
 local UIManager = require("ui/uimanager")
 local WidgetContainer = require("ui/widget/container/widgetcontainer")
 local logger = require("logger")
```
Comment (Frenzie) [inline plugins/terminal.koplugin/main.lua:13]: Seems lfs should go here? :-)

**Comment by poire-z** [comment]
Can you provide some screenshots?

**Comment by smartscripts-nl** [comment]
> Can you provide some screenshots?

See attachments. Partly in Dutch.
Reader_2020-08-31_104406
Reader_2020-08-31_104415

**Comment by smartscripts-nl** [comment]
> > Can you provide some screenshots?
> 
> See attachments. Partly in Dutch.

"Opslaan" = "Save"

**Comment by smartscripts-nl** [comment]
> Can you provide some screenshots?

Reader_2020-08-31_105109
Reader_2020-08-31_105055

"Verwijder" = "Remove"

**Comment by smartscripts-nl** [comment]
> Can you provide some screenshots?

Reader_2020-08-31_110525

"Terug" = "Back", "Sluiten" = "Close"

**Comment by Frenzie** [comment]
Hm, did I come up with "opdrachtuitvoer"? Looking at it on that screenshot I'm not the biggest fan of how that sounds.

**Comment by poire-z** [comment]
I hope you're not satisfied with the keyboard covering the bottom buttons, and the title overflowing screen top :) 
Try to not provide text_height (it should then adjust to text), or provide fullscreen=true.

**Comment by smartscripts-nl** [comment]
> Probably. Or just `git rebase master` to get up to date with your #6599 that has already been merged.
> And before bringing more PRs (that all start from master before #6599), can you finish your 2 other ones #6598 and #6600 please.

Finish #6598 and #6600: done.

**Comment by smartscripts-nl** [comment]
> I hope you're not satisfied with the keyboard covering the bottom buttons, and the title overflowing screen top :)
> Try to not provide text_height (it should then adjust to text), or provide fullscreen=true.

I've just borrowed the original code. I don't know what happens on other devices when I change the dimensions: maybe it then looks good on my Kobo Forma, but not on other devices?

**Comment by smartscripts-nl** [comment]
> Hm, did I come up with "opdrachtuitvoer"? Looking at it on that screenshot I'm not the biggest fan of how that sounds.
> 
> Edit: I guess it's the standard translation.

Indeed, this is the standard translation.

**Comment by Frenzie** [comment]
Wfm, @NiLuJe ?

**Comment by smartscripts-nl** [comment]
> Probably. Or just `git rebase master` to get up to date with your #6599 that has already been merged.
> And before bringing more PRs (that all start from master before #6599), can you finish your 2 other ones #6598 and #6600 please.

Is it acceptable that I discontinue this PR and start a fresh one, from upstream/master? I fear all kind of weird situations will come into existence when I try to fix things, because I already pushed my updates. Or maybe I'll try to fix things via the "Resolve conflicts" button.

**Comment by Frenzie** [comment]
> Is it acceptable 

That's fine, as long as you make sure you have your code you can resolve anything by force pushing in the worst case scenario though. ;-)

**Pull request closed without merging**